src/connection.js

```diff
@@ -207,12 +214,14 @@ const GetEpochScheduleRpcResult = struct({
 /**
  * Expected JSON RPC response for the "getBalance" message
  */
-const GetBalanceRpcResult = struct({
-  jsonrpc: struct.literal('2.0'),
-  id: 'string',
-  error: 'any?',
-  result: 'number?',
-});
+const GetBalanceWithCommitmentRpcResult = jsonRpcResult(
```

Comment:
It would also be nice if it was implemented in a backwards compatible way with an issue / comment to remove when your rpc changes get released in 0.21.

Something like (untested)
```
function jsonRpcResultWithContext(value: any) {
  return jsonRpcResult(struct.union([
    value,
    struct({
      context: struct({
        slot: 'number',
      }),
      value,
    }),
  ]);
}
```

Reply:
this is neat and I love the backwards compatibility idea, one issue I see is that Envelope<T> isn't a T. So I think we need to have a different method name, dispatch based on the presence of commitment in the input argument, or something like that. Let me code the dispatch idea and see what you all think.